### PR TITLE
ICDS Report Updates

### PR DESCRIPTION
**Pull request by sheelio**
Might be easier to read commit by commit, but this does a few things:
- Small bug fixes to the existing UCR reports and data sources
- Adds a new column to one of the data sources
- Cleans up the aggregation script
- Updates the SQL aggregation so that new column is available in the reporting tables
- Updates the SQL aggregation script to calculate if a given AWC is launched / trained yet or not

(Hoping interrupt can give this a quick review - @NoahCarnahan)

### CODE REVIEW

File: custom/icds_reports/ucr/expressions.py [hunk 1]
```diff
@@ -52,6 +56,52 @@ class GetLastFormRepeatSpec(JsonObject):
...
+    def configure(self, case_id_expression, case_forms_expression):
+        self._case_id_expression = case_id_expression
+        self._case_forms_expression = case_forms_expression
+
+    def __call__(self, item, context=None):
+        case_id = self._case_id_expression(item, context)
+        if not case_id:
+            return []
+
+        forms = self._case_forms_expression(item, context)
```
Comment (NoahCarnahan) [inline custom/icds_reports/ucr/expressions.py]: It looks like this line could come after you check for a cached value.

**Review by NoahCarnahan — CHANGES_REQUESTED** [review]
Looks good, just one small optimization I saw. Have you done all the testing you need to do on all of these changes?

File: custom/icds_reports/ucr/expressions.py [hunk 2]
```diff
@@ -52,6 +56,52 @@ class GetLastFormRepeatSpec(JsonObject):
...
+        case_id = self._case_id_expression(item, context)
+        if not case_id:
+            return []
+
+        cache_key = (self.__class__.__name__, case_id)
+        if context.get_cache_value(cache_key) is not None:
+            return context.get_cache_value(cache_key)
+
+        forms = self._case_forms_expression(item, context)
+
```
Comment (dimagimon) [inline custom/icds_reports/ucr/expressions.py]: W293 blank line contains whitespace

**Review by NoahCarnahan — APPROVED** [review]
👍

**Pull request merged**